libc-bottom-half/crt/crt1.c

```diff
@@ -110,18 +64,13 @@ void _start(void) {
         }
     }
 
-    // Fill in the arguments from WASI syscalls.
-    size_t argc;
-    char **argv;
-    if (populate_args(&argc, &argv) != __WASI_ESUCCESS) {
-        _Exit(EX_OSERR);
-    }
-
     // The linker synthesizes this to call constructors.
     __wasm_call_ctors();
 
-    // Call main with the arguments.
-    int r = main(argc, argv);
+    // Call `__original_main` which will either be a compiler-synthesized
+    // function which calls `main` with no arguemnts, or a libc routine
+    // which populates `argv` and `argc` and calls `main` with them.
```

Comment:
Is this comment accurate?

Is this more like.. ".. will either be the application's zero-argument main function (renamed by the compiler) or a libc routine which populates `argv` and `argc` and calls the application's two-argument `main`."

Reply:
Yeah, I guess I was trying to keep it abstract here, but it isn't essential. I've now updated the comment to your suggested wording.